src/components/users/UserMatches.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { object } from 'prop-types';
+import { useIntl } from 'react-intl';
+
+import 'style/App.scss';
+import MatchItem from 'components/common/MatchItem';
+
+const UserMatches = ({ userMatches }) => {
```

Comment:
```suggestion
const UserMatches = ({ userMatches = [] }) => {
```
You can add default values to props, so you don't have to validate for existence then. You can delete first userMatches at the beginning of line 13. Take into consideration that default values will be applied only when the func/componente receives and _undefined_ value for that prop, if it gets _null_ the prop won't take the default value and will rise an exception at the moment to evaluate userMatches.length.